test/cpp/end2end/async_end2end_test.cc

```diff
@@ -67,10 +67,10 @@ namespace {
 void* tag(int i) { return (void*)(intptr_t)i; }
 
 #ifdef GPR_POSIX_SOCKET
-static int assert_non_blocking_poll(struct pollfd* pfds, nfds_t nfds,
+static int non_blocking_poll(struct pollfd* pfds, nfds_t nfds,
                                     int timeout) {
-  GPR_ASSERT(timeout == 0);
-  return poll(pfds, nfds, timeout);
+  /* ignore timeout and always use timeout 0 */
```

Comment:
So the issue was that this function gets invoked not only from
async_end2end_test , but also from the resolver since the poll override now
takes place before the library really gets started working. This wasn't the
case before because the poll function was constantly getting overridden and
re-defaulted on each verification.

On Tue, Feb 16, 2016 at 10:03 AM Craig Tiller <notifications@github.com>
wrote:

> In test/cpp/end2end/async_end2end_test.cc
> <https://github.com/grpc/grpc/pull/5258#discussion_r53050942>:
>
> >                                      int timeout) {
> > -  GPR_ASSERT(timeout == 0);
> > -  return poll(pfds, nfds, timeout);
> > +  /* ignore timeout and always use timeout 0 */
>
> Removing the assert makes this a much weaker test.
>
> The original test provided guarantees that we never blocked calling code,
> which is important for many use cases (imagine a UI thread that wants to do
> *some* grpc work).
>
> Overriding that here means that we're no longer providing such a guarantee.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/5258/files#r53050942>.
>